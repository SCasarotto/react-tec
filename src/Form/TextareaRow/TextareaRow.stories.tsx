import React from 'react';

import { Story, Meta } from '@storybook/react';

import { TextareaRow, TextareaRowProps } from './TextareaRow';

export default {
  title: 'Form/TextareaRow',
  component: TextareaRow,
  argTypes: {},
} as Meta;

const Template: Story<TextareaRowProps> = (args) => <TextareaRow {...args} />;

export const Default = Template.bind({});
Default.args = {
  title: 'My Text Area',
  labelForKey: 'SomeUniqueKey42',
  onChange: (e) => console.log(e.target.value),
};
