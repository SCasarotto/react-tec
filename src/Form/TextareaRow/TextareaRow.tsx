import React from 'react';

import { FormRow, FormRowCustomProps } from '../FormRow';
import { Label, LabelCustomProps } from '../Label';
import { Textarea, TextareaProps } from '../Textarea';

export interface TextareaRowProps
  extends FormRowCustomProps,
    LabelCustomProps,
    TextareaProps {
  labelForKey: string;
}
/**
 *
 * A component containing a `Row`, a `Label` and either a `Textarea`. In addition to the props below, reference all props for `Textarea`.
 *
 */
export const TextareaRow = React.forwardRef<
  HTMLTextAreaElement,
  TextareaRowProps
>((props, ref) => {
  const {
    rowSize,
    last,
    className = '',
    title,
    onChange,
    disabled,
    value,
    required,
    labelForKey,
    ...rest
  } = props;

  return (
    <FormRow
      rowSize={rowSize}
      last={last}
      className={`tec-inputRow ${className}`}
    >
      {title && (
        <Label
          htmlFor={labelForKey}
          required={required}
          disabled={disabled}
          className='tec-inputRowTitle'
        >
          {title}
        </Label>
      )}
      <Textarea
        value={value}
        onChange={onChange}
        id={labelForKey}
        disabled={disabled}
        className='tec-inputRowInput'
        ref={ref}
        {...rest}
      />
    </FormRow>
  );
});
