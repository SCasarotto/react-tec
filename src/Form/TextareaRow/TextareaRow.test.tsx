import React from 'react';

import { render, waitFor } from '@testing-library/react';

import { TextareaRow } from './TextareaRow';

describe('TextareaRow', () => {
  // Renders Without Error
  it('matches snapshot', () => {
    const { asFragment } = render(
      <TextareaRow labelForKey='testTextareaRow' />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Class Name
  it('accepts custom className', () => {
    const { container } = render(
      <TextareaRow labelForKey='testTextareaRow' className='myClassName' />,
    );
    expect(container.firstChild).toHaveClass('myClassName');
  });

  // Ref
  it('renders with ref', async () => {
    const ref: { current: HTMLTextAreaElement | null } = { current: null };
    const { getByTestId } = render(
      <TextareaRow
        labelForKey='testTextareaRow'
        ref={ref}
        data-testid='myInput'
      />,
    );
    const input = getByTestId('myInput');
    await waitFor(() => expect(input).not.toHaveFocus());
    ref.current?.focus();
    await waitFor(() => expect(input).toHaveFocus());
  });
});
