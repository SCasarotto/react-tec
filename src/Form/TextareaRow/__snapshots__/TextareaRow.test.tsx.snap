// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextareaRow matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: relative;
  display: inline-block;
  vertical-align: top;
  margin-bottom: 10px;
  text-align: left;
  box-sizing: border-box;
  padding-right: 0px;
  width: 100%;
}

.c1 {
  position: relative;
  display: block;
  width: 100%;
  font-size: 14px;
  box-sizing: border-box;
  padding: 10px;
  min-height: 140px;
  border-radius: 5px;
  line-height: 1.5;
  margin: 0px;
  border: 1px solid;
  -webkit-transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
  transition: border-color 0.2s ease-in,box-shadow 0.2s ease-in;
}

.c1:hover,
.c1:active {
  border: 1px solid;
  box-shadow: 0 0 0 1px inset;
}

.c1:disabled {
  border: 1px solid;
  box-shadow: none;
  cursor: not-allowed;
}

<div
    class="c0 tec-formRow tec-inputRow "
  >
    <textarea
      class="c1 tec-textarea tec-inputRowInput"
      id="testTextareaRow"
    />
  </div>
</DocumentFragment>
`;
